cameraview/src/main/java/com/otaliastudios/cameraview/CameraView.java

```diff
@@ -280,6 +323,26 @@ protected void onDetachedFromWindow() {
         super.onDetachedFromWindow();
     }
 
+    @Override
+    protected boolean checkLayoutParams(ViewGroup.LayoutParams p) {
```

Comment:
Will tell you how this should be IMO:

- `checkLayoutParams(ViewGroup.LayoutParams p)` should be removed and left to super
- `generateDefaultLayoutParams()` should also be removed
- `generateLayoutParams(ViewGroup.LayoutParams p)` should check if p is OverlayLayoutParams, in that case return p. If not, return super.
- `generateLayoutParams(AttributeSet attributeSet)` should return an `OverlayLayoutParams` if the attributeSet has isOverlay set to true

This way it is much simpler and we leave CameraView with its own default parameters. We just return OverlayLayoutParams when isOverlay=true.